spicy/toolchain/src/compiler/codegen/parser-builder.cc

```diff
@@ -974,9 +1021,34 @@ struct ProductionVisitor
             pushState(std::move(pstate));
         }
 
-        for ( const auto& i : p.fields() )
+        if ( const auto& skipPre = p.unitType().propertyItem("%skip-pre") ) {
+            auto ctor = skipPre->expression()->as<hilti::expression::Ctor>().ctor();
+            if ( auto c = ctor.tryAs<hilti::ctor::RegExp>() )
+                skipRegExp(*c);
+        }
```

Comment:
Could compress this (and below) if `skipRegExp` took an `Expression` and did the cast and null check there.